### PR TITLE
Vision Transforms

### PR DESCRIPTION
**Pull request by NiklasGustafsson**
This PR adds a number of vision transforms from torchvision. It's a big PR, but mostly a repetition of a few patterns.

### CODE REVIEW

File: src/TorchSharp/Tensor/TensorExtensionMethods.cs [hunk 1]
```diff
@@ -335,5 +340,28 @@ public static double clip_grad_norm(this IList<TorchTensor> tensors, double max_
...
+        public static TorchTensor crop(this TorchTensor image, int top, int left, int height, int width)
+        {
+            var dims = image.Dimensions;
+            var hoffset = dims - 2;
+            long h = image.shape[hoffset], w = image.shape[hoffset + 1];
+
+            var right = left + width;
+            var bottom = top + height;
+
+            if (left < 0 || top < 0 || right > w || bottom > h) {
```
Comment (dsyme) [inline src/TorchSharp/Tensor/TensorExtensionMethods.cs:355]: Maybe put a comment here about why this protection is needed in C# (doesn't the C++ raise a decent exception?)

File: src/TorchSharp/Tensor/TensorExtensionMethods.cs [hunk 2]
```diff
@@ -335,5 +340,28 @@ public static double clip_grad_norm(this IList<TorchTensor> tensors, double max_
...
+
+                //var padding_ltrb = new long[] { Math.Max(-left, 0), Math.Max(-top, 0), Math.Max(right - w, 0), Math.Max(bottom - h, 0) };
```
Comment (dsyme) [inline src/TorchSharp/Tensor/TensorExtensionMethods.cs:357]: Add a comment on why this code is commented out?

File: src/FSharp.Examples/SequenceToSequence.fs [hunk 3]
```diff
@@ -60,8 +60,8 @@ type PositionalEncoding(dmodel, maxLen) as this =
         let divTerm = (Float32Tensor.arange(0L.ToScalar(), dmodel.ToScalar(), 2L.ToScalar()) * (-Math.Log(10000.0) / (float dmodel)).ToScalar()).exp()
 
         let NULL = System.Nullable<int64>()
-        pe.[TorchTensorIndex.Ellipsis, TorchTensorIndex.Slice(0L, NULL, 2L)] <- (position * divTerm).sin()
```
[the remark this answers is not in the document]
Comment (dsyme) [reply src/FSharp.Examples/SequenceToSequence.fs:63]: A workaround is to use an explicit array, e.g. either of 

```fsharp
        pe.[ [| TorchTensorIndex.Ellipsis; TorchTensorIndex.Slice(0L, NULL, 2L) |] ] <- (position * divTerm).sin()
```
or the named property setter: 
```fsharp
        pe.Item( [| TorchTensorIndex.Ellipsis; TorchTensorIndex.Slice(0L, NULL, 2L) |] ) <- (position * divTerm).sin()
```
or
```fsharp
        pe.set_Item( [| TorchTensorIndex.Ellipsis; TorchTensorIndex.Slice(0L, NULL, 2L) |], (position * divTerm).sin())
```

**Review by dsyme — APPROVED** [review]

File: src/TorchSharp/Tensor/TensorExtensionMethods.cs (hunk 1 above)
Comment (NiklasGustafsson) [reply src/TorchSharp/Tensor/TensorExtensionMethods.cs:355]: Good point. I'll fix this up. No reason to check in commented code.

File: src/TorchSharp/Tensor/TensorExtensionMethods.cs (hunk 2 above)
Comment (NiklasGustafsson) [reply src/TorchSharp/Tensor/TensorExtensionMethods.cs:357]: Yes.

File: src/FSharp.Examples/SequenceToSequence.fs (hunk 3 above)
[the remark this answers is not in the document]
Comment (NiklasGustafsson) [reply src/FSharp.Examples/SequenceToSequence.fs:63]: I'll leave it as is for now,

[the remark this answers is not in the document]
Comment (dsyme) [reply src/FSharp.Examples/SequenceToSequence.fs:63]: @Niklas the limitation for ParamArray index setters is here: https://github.com/dotnet/fsharp/issues/9369

[the remark this answers is not in the document]
Comment (NiklasGustafsson) [reply src/FSharp.Examples/SequenceToSequence.fs:63]: @dsyme -- thanks for following up. We should hold off changing the code until this makes it into the VS distribution of F#, IMO

**Pull request merged**